tests/nested/manual/optee-fde/task.yaml

```diff
@@ -0,0 +1,200 @@
+summary: Test snapd OPTEE integration
+
+details: |
+    Test that snapd will use a well known OPTEE TA (trusted application) as the
+    sealing method for FDE. The snaps used here are custom snaps that help
+    exercise this functionality. Their sources can be found in
+    github.com/canonical/optee-uc-fde.
+
+systems: [ubuntu-24.04-arm-64]
+
+environment:
+    NESTED_BUILD_SNAPD_FROM_CURRENT: true
+    NESTED_REPACK_KERNEL_SNAP: false
+    NESTED_REPACK_GADGET_SNAP: false
+    NESTED_REPACK_BASE_SNAP: false
+    NESTED_ENABLE_ARM_TRUSTZONE: true
+    NESTED_CPUS: 4
+    NESTED_MEM: 4096
+    NESTED_FAKESTORE_BLOB_DIR: $(pwd)/fake-store-blobdir
+    NESTED_SIGN_SNAPS_FAKESTORE: true
+    NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL: http://localhost:11028
+
+    KEY_PROTECTOR/optee: "optee"
+    KEY_PROTECTOR/hooks: "fde-setup-hook"
+
+prepare: |
+    if [ "${TRUST_TEST_KEYS}" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    kernel_name=""
+    kernel_revision=""
+    gadget_name=""
+    gadget_revision=""
+    if [ "${KEY_PROTECTOR}" = "optee" ]; then
+        kernel_name="test-qemu-optee-kernel"
+        kernel_revision=2
+        gadget_name="test-qemu-optee-gadget"
+        gadget_revision=5
+    else
+        kernel_name="test-qemu-fde-hooks-kernel"
+        kernel_revision=2
+        gadget_name="test-qemu-fde-hooks-gadget"
```

Comment:
It is arm-trusted + u-boot for qemu. Looking at the snaps, maybe at least the description of those snaps should mention it. It seems to say for both fde-hooks and optee: "Gadget for QEMU & OP-TEE integration testing".

Reply:
(not related to this PR)